### ISSUE TITLE
Global environment variables

### ISSUE THREAD

**Issue by skatkar** [rangav/thunder-client-support#70]
**Is your feature request related to a problem? Please describe.**
I am trying to set the variable which is common across all the environments. This is what I was using in Postman
`var jsonData = JSON.parse(responseBody);
postman.setEnvironmentVariable("access_token", jsonData.access_token);
`
This script was taking care of adding that variable to whatever environment I have selected. But in Thunder client, I need to create the variable with an empty value for each environment and then I can set the value from the response body.

**Describe the solution you'd like**
Global variable support and a way to set the value.

**Describe alternatives you've considered**

**Implementation:**

**Comment by wildhart** [comment]
This would also be useful for having environment variables which are common across environments.  E.g. you could store just the path/userId/token data in DEV/TEST/LIVE environment groups, and then store common data in a shared group (e.g. shared test data).

**Comment by rangav** [comment]
Hi @skatkar and @wildhart will change set env var to create variable if missing in selected env.

**Comment by RealSollyM** [comment]
Hi @rangav . Global Env variables would actually be awesome. In my case, the only difference between my environments is the server address (mostly), the rest of the information is the same. Like @skatkar I do store the access_token back into the Global Env Var but didn't know it was possible to do it programmatically. This would indeed speed up a lot of things. If you could allow one of the Env to be marked as global, then a var that is not in the active Env would then be searched from the Global Env, that would be great. Thanks for an awesome work so far.

**Comment by rangav** [comment]
HI all, can you guys look into these other issues similar to this, and see what option best suits everyone
https://github.com/rangav/thunder-client-support/issues/74, https://github.com/rangav/thunder-client-support/issues/79

**Comment by wildhart** [comment]
I think, in addition to my comment in #74 (using .env files) which would be useful for some projects, it would still be good to have multiple ENV files active with one as the default, since I don't want to pollute my .env files with environment-specific test data.

I think it could work if you put a checkbox to the left of each environment so they could be individually turned on/off, combined with the existing menu option to set one as the default.

**Comment by skatkar** [comment]
On the similar line of #74 -> To simplify global environment variables, environment-specific files, etc; I want to suggest an approach. 

1. Create a single environment file in which the user will mention which variables he wants.
2. Now there is a trick about the format of this file
```
[global]
variable_1
variable_2
[dev]
variable_3
variable_4
[prod]
variable_3
variable_4
```
3. As mentioned above, let the user specify which variables he wants for a specific environment. Also, allow the user to declare which environment he wants.
4. If the user forgets to provide `[environment_name]` then use that variable at a global level.
5. This will give the user full control over declaring variables and the environment names as well.
6. If we don't want the user to upload such a file then https://github.com/rangav/thunder-client-support/issues/70#issuecomment-827291310 this approach looks better.

**Comment by rangav** [comment]
Hi @RealSollyM @wildhart @wildhart 

I think  https://github.com/rangav/thunder-client-support/issues/70#issuecomment-830100080 is a simple and clear solution as suggested by @RealSollyM. That should be easy to implement and easy to use for most users.

I will update on this when I start development.

**Comment by nghtstr** [comment]
Might I make the following suggestion then, as it seems to me a way to handle the different types of variables being proposed. I would suggest having a pre-defined global set (it could reside in the Env tab, because it makes sense), but having that as the last fallback. I would suggest having collection specific variables as well. This would add the most amount of flexibility to this fantastic plugin!

So that order of operations would be: Collection > Environment > Global.

You could have Folder specific variables, but that might be going too far down the rabbit hole.

**Comment by rangav** [comment]
Thanks @nghtstr for the suggestions

**Comment by rangav** [comment]
Global variables feature will be part of next update

**Comment by ajaydwarkani** [comment]
@rangav , how about adding support for the system environment variable that can be used as global variables?

**Comment by rangav** [comment]
Hi @ajaydwarkani did not understood your question, can you please provide more details or an example

**Comment by ajaydwarkani** [comment]
@rangav , What I mean to add support for using OS system environment variable. 
For example, if I have an OS system variable as `API_TOKEN=some-value` then I should simply be able to consume it in thunder client as `{{API_TOKEN}}` or `{{#API_TOKEN}}` without needing to import and declaring it.

**Comment by rangav** [comment]
That's a good suggestion, can you please create new feature request for this, we can track there.

**Comment by ajaydwarkani** [comment]
@rangav here is the feature request https://github.com/rangav/thunder-client-support/issues/276

**Comment by rangav** [comment]
This feature is now implemented in v1.8.0

see all features in v1.8.0
https://github.com/rangav/thunder-client-support/releases/tag/v1.8.0
